### ISSUE TITLE
timeout running zos_job_submit

### ISSUE THREAD

**Issue by petelouis** [ansible-collections/ibm_zos_core#245]
I have a playbook that submits 2 jobs via zos_job_submit. Both jobs complete successfully on the z/OS system. Ansible is able to successfully retrieve results from the first job, but for the second it returns the following error:

fatal: [tvt5033.svl.ibm.com]: FAILED! => {"changed": false, "job_id": "JOB00994", "msg": "SubmitJCLError('THE JOB CAN NOT BE QUERIED FROM JES (TIMEOUT=10s). PLEASE CHECK THE ZOS SYSTEM. IT IS SLOW TO RESPONSE.')"}

I can't find any way to adjust the 10 second timeout value, am I missing something? Also, I'm able to run the same playbook against a different z/OS system without seeing this timeout error.

**Comment by blakeinate** [comment]
Hi @petelouis , would you be able to share what version of the ibm_zos_core collection you are using?

**Comment by petelouis** [comment]
Not sure of the version, is there a way to check?

**Comment by petelouis** [comment]
If I look in zos_job_submit.py, I see:
ANSIBLE_METADATA = {
    "metadata_version": "1.1",
    "status": ["preview"],
    "supported_by": "community",
}

**Comment by richp405** [comment]
At a uss command prompt type zoaversiion, and it should give a number and
date.  I think I understand the 10 second issue, and it may become
configurable.

**Comment by richp405** [comment]
Typo... zoaversion.  Just one I.

**Comment by petelouis** [comment]
zoaversion shows 2020/07/07 14:54:31 CUT V1.0.3   

Also, I tried increasing the timeout value in zos_job_submit.py, but got the same result, so there may be something else going on there.

**Comment by richp405** [comment]
I agree with you, the delay is a core issue.  There is a newer zoau, but it
won't impact this issue.  Could you send the actual playbook where this
call is?  I'm curious if you're sending one call or two to job submit.

**Comment by petelouis** [comment]
Here is the main playbook:
---

- name: db2 setup
  # hosts: tvt5023.svl.ibm.com
  hosts: TVT5033
  gather_facts: yes
  vars_files:
    - tvt5033.yml
  environment:
    PATH: "/u/louis/.ansible:/usr/lpp/IBM/cyp/v3r8/pyz/bin:/bin:/usr/lpp/java/J8.0_64/bin:/usr/lpp/db2c10/jdbc/bin:/usr/lpp/IBM/zoautil/bin" 
    LIBPATH: "/usr/lib/java_runtime:/usr/lpp/db2c10/base/lib:/usr/lpp/IBM/cyp/v3r8/pyz/lib:/usr/lpp/IBM/zoautil/lib" 
    PYTHONPATH: "/usr/lpp/IBM/cyp/v3r8/pyz/lib:/usr/lpp/IBM/zoautil/lib" 
    CLASSPATH: "/usr/lpp/IBM/zoautil/lib"
    _BPXK_AUTOCVT: "ON"
    _CEE_RUNOPTS: "FILETAG(AUTOCVT,AUTOTAG) POSIX(ON)"
    _TAG_REDIR_ERR: "txt"                                
    _TAG_REDIR_IN: "txt"
    _TAG_REDIR_OUT: "txt"

  collections:
    - ibm.ibm_zos_core

  tasks:
    - name: define db2 alias
      include_tasks: tasks/doc/db2_alias.yml

    - name: define db2 dataclas
      include_tasks: tasks/doc/db2_dc.yml

and here are the 2 yml files from the tasks section above:

**db2_alias.yml:**
---

- name: get date on server
  shell: date
  register: output

- name: upload jcl for alias job
  template:
    src: db2_alias.j2
    dest: /tmp/db2_alias
    backup: no

- name: change encoding
  zos_encode:
    src: /tmp/db2_alias
    dest: /tmp/db2_alias.jcl
    from_encoding: ISO8859-1
    to_encoding: IBM-1047
    backup: no

- name: run alias job
  zos_job_submit:
    src: "/tmp/db2_alias.jcl"
    location: USS
    wait: true
    return_output: true
  register: job_detail

- name: setting fact 'job results'
  set_fact:
    job_results: "{{ job_detail.jobs[0].ddnames[3].content | join('\n') }}"

- name: alias defined? 
  set_fact:
    alias_defined: true
  when: "alias_success in job_results"       

- name: alias defined previously? 
  set_fact:
    alias_defined_previously: true
  when: "alias_dup in job_results"       

- name: display variables
  debug: 
    msg: "alias_defined = {{alias_defined}}, alias_defined_previously = {{alias_defined_previously}}"

============================================

**db2_dc.yml:**
---

- name: get date on server
  shell: date
  register: output

- name: upload jcl for sms validate job
  template:
    src: db2_smsval.j2
    dest: /tmp/smsval
    backup: no

- name: change encoding
  zos_encode:
    src: /tmp/smsval
    dest: /tmp/smsval.jcl
    from_encoding: ISO8859-1
    to_encoding: IBM-1047
    backup: no

- name: run sms validate job
  zos_job_submit:
    src: "/tmp/smsval.jcl"
    location: USS
    wait: true
    wait_time_s: 60
    return_output: true

**Comment by richp405** [comment]
I believe this is related to bugfix/3687, which I'll be re-opening on Tuesday after finishing a test of the timing issue.

**Comment by petelouis** [comment]
I've investigated this further, and now it appears that the problem may have occurred because my zoau config was not quite complete.  After updating a couple more envvars and defining some SDSF RACF profiles on my z/OS system, it looks like the zos_job_submit error is fixed.

**Comment by ddimatos** [comment]
Just a few comments to make; 
- Blake was asking for the `ibm_zos_core` collection, that can be determined by running the command documented [here](https://ibm.github.io/z_ansible_collections_doc/faqs/faqs.html#how-can-i-know-which-version-of-a-collection-i-am-using)
- @richp405 did find a but in `zos_job_submit` related to timing, which was released in our latest [v1.3.0-beta.2](https://galaxy.ansible.com/ibm/ibm_zos_core) today
- It appears from your @petelouis last comment that your OMVS segment/profile was edited and allowed the ansible user enough authority to access the submitted job. 

Guessing we can close this now, feel free to reopen and try out the latest release. 
Thank you for reporting this.
